zammad/templates/_helpers.tpl

```diff
@@ -63,54 +63,77 @@ Create the name of the service account to use
 {{- end -}}
 
 {{/*
-autowizard secret name
+autowizard config
 */}}
-{{- define "zammad.autowizardSecretName" -}}
+{{- define "zammad.env.AUTOWIZARD_JSON" -}}
+- name: AUTOWIZARD_JSON
 {{- if .Values.secrets.autowizard.useExisting -}}
-{{ .Values.secrets.autowizard.secretName }}
+  valueFrom:
+    secretKeyRef:
+      key: {{ .Values.secrets.autowizard.secretKey }}
+      name: {{ .Values.secrets.autowizard.secretName }}
 {{- else -}}
-{{ template "zammad.fullname" . }}-{{ .Values.secrets.autowizard.secretName }}
+  value: {{ .Values.autoWizard.config | b64enc | quote }}
```

Comment:
I don't see the improvment here.
Before we had a secret reference.
Now we have a plaintext secret.
From a security perpective, thats the less secure option, as exposure of the secret can happen more easy (logs, process lists or other containers in the same process namespace can read it for example).

Reply:
@monotek can you have another look, please? I reset the auto wizard handling to the previous state. Is it acceptable now? I would still want to add a test case for the auto wizard handling to the new 'full' scenario.

Can you please briefly explain what you mean by 'less secure' here? The auto wizard data still ends up in an ENV var of the container, no matter if there is an underlying secret or not. Are you referring to other kubernetes processes which might read the plaintext content via API?

Reply:
It also applies for the other secrets.

Yes, its true that it's also an env var in the end. To prevent that we would need to use a volume mount of the secret instead using an env var.

With a plaintext secret stuff like helm template would still suffer, as the secret would be printed out directly.

Reply:
So what's your suggestion @monotek. Drop this entire MR (except for the test improvements) and close the bug as `wontfix`?

TBH I'm not sure if that's the right way. When people use existing secrets, everything is fine as before (there the security implication is not being able to use special characters, which is also not good). Only for local values from `values.yaml`, the behaviour would be different.

It seems strange that it should not be possible to provide URL encoded ENV vars via k8s in a proper way.

Reply:
I don't now if there is a solution but i also don't have the time currently to check that out.
Sorry :(
I just would not like to see a solution introduced which might have security implications.

Reply:
All secrets changes reverted.